views/js/qtiCreator/plugins/menu/preview.js

```diff
@@ -55,32 +109,31 @@ define([
              * @event itemCreator#preview
              * @param {String} uri - the uri of this item to preview
              */
-            itemCreator.on('preview', function(uri){
-              	var type = 'qtiItem';
-
-                previewerFactory(type, uri, { }, {
-                    readOnly: false,
-                    fullPage: true
-                });
+            itemCreator.on('preview', function(uri) {
+                if (!this.isEmpty()) {
+                    var type = 'qtiItem';
+
+                    previewerFactory(type, uri, {}, {
+                        readOnly: false,
+                        fullPage: true
+                    });
+                }
             });
 
+            itemCreator.on('saved', () => enablePreviewIfNotEmpty(this));
+
             //creates the preview button
             this.$element = $(buttonTpl({
                 icon: 'preview',
                 title: __('Preview the item'),
                 text : __('Preview'),
                 cssClass: 'preview-trigger'
-            })).on('click', function previewHandler(e){
-                $(document).trigger('open-preview.qti-item');
-
-                e.preventDefault();
+            })).on('click', e => previewHandler(e, this));
 
-                self.disable();
-
-                itemCreator.trigger('preview', itemCreator.getItem().data('uri'));
-
-                self.enable();
-            });
+            this.getAreaBroker()
+                .getItemPanelArea()
+                .on('dropped.gridEdit.insertable', () => disablePreviewIfEmpty(this))
```

Comment:
Do we need this? When smth is inserted condition `if (that.getHost().isEmpty()) {` will be false

Reply:
Will be false except if you remove all interactions. If you remove all interactions, preview is disabled. If some interaction remains you can go to preview, appear popup for save or not (doesn't matter because in both cases you will have some content)

Reply:
Done